src/charm.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+#
+# Learn more at: https://juju.is/docs/sdk
+
+"""Integrator charm to provide a fiveg_gnb_identity."""
+
+import logging
+from typing import Optional, cast
+
+import ops
+from charms.sdcore_gnbsim_k8s.v0.fiveg_gnb_identity import (  # type: ignore[import]
+    GnbIdentityProvides,
+)
+from ops import ActiveStatus, BlockedStatus, CollectStatusEvent
+
+logger = logging.getLogger(__name__)
+
+GNB_IDENTITY_RELATION_NAME = "fiveg_gnb_identity"
+
+
+class SdcoreGnbIntegratorCharm(ops.CharmBase):
+    """Charm for gNB Integrator Service."""
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.framework.observe(self.on.collect_unit_status, self._on_collect_unit_status)
+        self.framework.observe(self.on.config_changed, self._on_config_changed)
+        self._gnb_identity_provider = GnbIdentityProvides(self, GNB_IDENTITY_RELATION_NAME)
+        self.framework.observe(
+            self._gnb_identity_provider.on.fiveg_gnb_identity_request,
+            self._on_config_changed,
+        )
+
+    def _on_collect_unit_status(self, event: CollectStatusEvent):
+        """Checks the unit status and set it when CollectStatusEvent is fired.
+
+        Args:
+            event: CollectStatusEvent
+        """
+        if invalid_configs := self._get_invalid_configs():
+            event.add_status(
+                BlockedStatus(f"The following configurations are not valid: {invalid_configs}")
+            )
+            logger.info("The following configurations are not valid: %s", invalid_configs)
+            return
+        event.add_status(ActiveStatus())
+
+    def _on_config_changed(self, event: ops.ConfigChangedEvent):
```

Comment:
This event handler is called on multiple events, not just `configchanged` we should change its name and the `event` should be more generic (EventBase)